src/get-started/java-binding.md

```diff
@@ -0,0 +1,382 @@
+    ```
+    def handle_tx(request):
+      var body = request.getBody()
+      try:
+        var message = txMessageFromBody(body)
+        var tx = service.convertTransaction(txMessage)
+        var hash = tx.hash()
+        node.submitTransaction(tx)
+        return ok(createTxResponse(hash))
+      catch IllegalArgumentException as e:
+        return badRequest("Invalid request: " + e)
+      catch VerificationError as e:
+        return badRequest("Invalid transaction: " + e)
+      catch Error as e:
+        log("Internal error:" + e)
+        return internalServerError()
+      ```
+
+#### Executable Transactions
+
+To correctly process a transaction, it must be transformed into an
+*executable transaction* (see `TransactionConverter` or
+`Service#convertTransaction` method) and transmitted to the framework using
+`Node#submitTransaction` method. The framework checks it, and if it is correct,
+broadcasts it to other nodes of the system. Other nodes, having received the
+transaction message, convert it into an executable transaction, also using the
+service transaction converter.
+
+An executable transaction is a class object that implements
+`com.exonum.binding.messages.Transaction` interface and defines transaction
+behavior and business logic. The interface implementations must define the
+transaction authentication rule (usually, the digital signature verification of
+the message) - `isValid` method; and the execution rule for the
+transaction - `execute` method.
+
+Ed25519 is a standard cryptographic system for digital signature of Exonum
+messages. We use `com.exonum.binding.crypto.CryptoFunctions#ed25519`
+method for obtaining the implementation of the function.
+
+The implementation of `Transaction#isValid` transaction authentication method
+must be a pure function, i.e. for the given transaction to return the same
+result on all nodes of the system. For this reason, access to the Exonum
+storage by external files or network resources in method implementations is
+not allowed.
+
+`Transaction#execute` method describes the operations that are applied to the
+current storage state when the transaction is executed. Exonum transfers `Fork`
+as an argument - a view that allows performing modifying operations. A service
+schema object can be used to access data collections of this service.
+
+!!! note
+    `Transaction#info` will be deleted soon.
+
+For more details about transactions in Exonum - their properties and processing
+rules - see our [documentation][transactions]
+
+### Description of the External Service API
+
+The external service API serves for interaction of the service and the external
+systems. A set of operations is defined by each service and can include sending
+transactions to the network, read requests for the blockchain data with the
+provision of the cryptographic proof, etc. Exonum provides an embedded web
+framework for describing the REST-interface of the service.
```

Comment:
"describing" probably isn't the right word here; there is no meta info on service endpoints thus far (which is what I'd think "describing the REST interface" means).